core/shared/src/main/scala-2.13+/fs2/Chunk213And3Compat.scala

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2013 Functional Streams for Scala
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package fs2
+
+import scala.collection.{Factory, SeqFactory}
+import scala.collection.immutable.ArraySeq
+import scala.reflect.ClassTag
+
+private[fs2] trait Chunk213And3Compat[+O] { self: Chunk[O] =>
+  def toArraySeq[O2 >: O: ClassTag]: ArraySeq[O2] = {
+    val array: Array[O2] = new Array[O2](size)
+    copyToArray(array)
+    ArraySeq.unsafeWrapArray[O2](array)
+  }
+
+  def toArraySeqUntagged: ArraySeq[O] = {
+    val buf = ArraySeq.untagged.newBuilder[O]
+    buf.sizeHint(size)
+    var i = 0
+    while (i < size) {
+      buf += apply(i)
+      i += 1
+    }
+    buf.result()
+  }
+
+  /** Views this Chunk as a Scala immutable Seq.
+    * Contrary to all methods that start with _"to"_ (e.g. {{toVector}}, {{toArray}}),
+    * this method does not copy data.
+    * As such, this method is mostly intended for `foreach` kind of interop.
+    */
+  def asSeq: Seq[O] =
+    new ChunkAsSeq(this)
```

Comment:
Perhaps it is worth some optimization:
As I can see, there is already
```
private final class IndexedSeqChunk[O](s: GIndexedSeq[O]) extends Chunk[O]
```
So if `this` is `IndexedSeqChunk` is would be better to return the underlying `s` from here.

Reply:
@satorg @armanbilge I tried to do this, but `GIndexedSeq` is `collection.IndexedSeq` whereas we are using `collection.immutable.Seq`

Personally, I find it weird and wrong that `Chunk` is using the general trait, since that allows mutable collections, meaning the data may change under the hood.

Reply:
yeah me too... I guess it might be due to an oversight....